packages/backend/src/open_payments/payment/outgoing/routes.ts

```diff
@@ -151,3 +133,13 @@ function outgoingPaymentToBody(
 ): OpenPaymentsOutgoingPayment {
   return outgoingPayment.toOpenPaymentsType(walletAddress)
 }
+
+function handleMissingWalletAddress(
+  deps: ServiceDependencies,
+  outgoingPayment: OutgoingPayment
+): never {
+  const errorMessage =
+    'Outgoing payment does not have wallet address. This should not be possible.'
+  deps.logger.error({ outgoingPaymentId: outgoingPayment.id }, errorMessage)
+  throw new Error(errorMessage)
```

Comment:
Just throwing normal error here since it's something quite unexpected, and shouldn't ever actually happen

Reply:
definitely agree with your reasoning here.

this is not a suggestion, just a stackoverflow post with some interesting food for thought regarding logging errors and throwing them: https://stackoverflow.com/questions/22072343/throw-exception-vs-logging

I did some similar work previously in cleaning up errors/logs and this is triggering flashbacks. One of the things I remember was our GCP error reports would include duplicate (but not obviously duplicate) errors because of this. Could be dependent on structured logger configuration and cloud error reporting/logging services. In these scenarios I changed them to warnings so they didn't show in error reports but could still be filtered in the log viewer. 

I'm just throwing out 1 anecdote and not claiming this is ideal. There are dissenting opinions in that post as well. Just thought it was relevant.

Reply:
Looks like I should also look at the middleware handling this...

I guess in this case it will:

- log at error level
- log again at error level in `openPaymentsServerErrorMiddleware` with some additional generic details
- throw generic 500

Reply:
@BlairCurrey definitely makes sense, and those order of events are correct. 

I think I would also prefer just to keep only the "main" log in the middleware, but I think the only reason I included this was because I also wanted to log the outgoingPaymentId, which otherwise wouldn't get logged. I don't have a nice solution of propagating that id out to the log in the middleware (particularly for the create routes)

Reply:
yeah... that seems fair. I dont think we can expect to log all the context in some central place away from where the actually error happens.